src/web/src/components/SearchResults.tsx

```diff
@@ -59,7 +59,9 @@ const SearchResults = () => {
   const [totalPosts, setTotalPosts] = useState(0);
 
   const prepareUrl = (index: number) =>
-    `${searchServiceUrl}?text=${encodeURIComponent(textParam)}&filter=${filter}&page=${index}`;
+    `${searchServiceUrl}?${filter === 'author' ? `author` : `post`}=${encodeURIComponent(
```

Comment:
I haven't tested this yet, but you could apply the logic for the filter in the Provider and just import it through the search hook, rather than doing this

Reply:
I rather not change it, because in the upcoming PRs for the advanced route, I'm converting it to something like:
```
${searchServiceUrl}/?author=${encodeURIComponent(author)}&post=${encodeURIComponent(
      post
    )}&title=${encodeURIComponent(title)}${from ? `&from=${from}` : ''}${to ? `&to=${to}` : ''}`;
```

We're completely removing out the filter. We can search stuff with empty fields so it's no longer this or that. It's a lot more flexibility.